web/app/js/components/NetworkGraph.jsx

```diff
@@ -0,0 +1,198 @@
+import _ from 'lodash';
+import { metricsPropType } from './util/MetricUtils.js';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withContext } from './util/AppContext.jsx';
+import withREST from './util/withREST.jsx';
+import * as d3 from 'd3';
+import 'whatwg-fetch';
+
+
+const defaultSvgWidth = 524;
+const defaultSvgHeight = 325;
+const margin = { top: 0, right: 0, bottom: 10, left: 0 };
+
+const simulation = d3.forceSimulation()
+  .force("link",
+    d3.forceLink()
+      .id(d => d.id)
+      .distance(140))
+  .force("charge", d3.forceManyBody().strength(-20))
+  .force("center", d3.forceCenter(defaultSvgWidth / 2, defaultSvgHeight / 2));
+
+export class NetworkGraphBase extends React.Component {
+  static defaultProps = {
+    deployments: []
+  }
+
+  static propTypes = {
+    data: PropTypes.arrayOf(metricsPropType.isRequired).isRequired,
+    deployments: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.svg = d3.select(".network-graph-container")
+      .append("svg")
+      .attr("class", "network-graph")
+      .attr("width", defaultSvgWidth)
```

Comment:
tioli we could make the svg the full width of the page:

```
    let container = document.getElementsByClassName("network-graph-container")[0];
    let width = !container ? defaultSvgWidth : container.getBoundingClientRect().width;
    

    this.svg = d3.select(".network-graph-container")
      .append("svg")
      .attr("class", "network-graph")
      .attr("width", width)
      .attr("height", defaultSvgHeight)
      .append("g")
      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");

    simulation.force("center", d3.forceCenter(width / 2, defaultSvgHeight / 2));
```

Note that we'd also need to center the simulation to the full width, by readjusting the centering here.